debian/patches/0060-add-simd-optimized-tonemapx-filter.patch

```diff
@@ -3874,38 +3896,35 @@ Index: FFmpeg/libavfilter/vf_tonemapx.c
 +            goto fail;
 +    }
 +
-+    if (s->apply_dovi) {
-+        AVFrameSideData *dovi_sd = av_frame_get_side_data(in, AV_FRAME_DATA_DOVI_METADATA);
-+        if (dovi_sd) {
-+            const AVDOVIMetadata *metadata = (AVDOVIMetadata *) dovi_sd->data;
-+            const AVDOVIRpuDataHeader *rpu = av_dovi_get_header(metadata);
-+            // only map dovi rpus that don't require an EL and has rpu profile == 0
-+            // for performance reason we only want to do reshaping when absolutely needed
-+            // such videos usually have vdr_rpu_profile == 0, for example profile 5 videos
-+            // this could be wrong as there is no public documentation on this field
-+            if (rpu->disable_residual_flag && rpu->vdr_rpu_profile == 0) {
-+                struct DoviMetadata *dovi = av_malloc(sizeof(*dovi));
-+                s->dovi = dovi;
-+                if (!s->dovi)
-+                    goto fail;
-+
-+                ff_map_dovi_metadata(s->dovi, metadata);
-+            }
++    if (dovi_sd) {
++        const AVDOVIMetadata *metadata = (AVDOVIMetadata *) dovi_sd->data;
++        const AVDOVIRpuDataHeader *rpu = av_dovi_get_header(metadata);
++        // only map dovi rpus that don't require an EL and has rpu profile == 0
++        // for performance reason we only want to do reshaping when absolutely needed
++        // such videos usually have vdr_rpu_profile == 0, for example profile 5 videos
++        // this could be wrong as there is no public documentation on this field
++        if (rpu->disable_residual_flag && rpu->vdr_rpu_profile == 0) {
++            struct DoviMetadata *dovi = av_malloc(sizeof(*dovi));
++            s->dovi = dovi;
++            if (!s->dovi)
++                goto fail;
++
++            ff_map_dovi_metadata(s->dovi, metadata);
```

Comment:
Reshaped input frames should be hardcoded as PQ, in case of incorrect setparams.
```c
in->color_trc = AVCOL_TRC_SMPTE2084;
in->colorspace = AVCOL_SPC_BT2020_NCL;
in->color_primaries = AVCOL_PRI_BT2020;
```

And this field should be executed earlier.
```
dovi
...
compute_lut
...
compute_coeffs
```